### PR TITLE
Feature/login and signup

### PR DESCRIPTION
**Pull request by faris114**
Built initial authentication function that will request email and password verification from the back-end

### CODE REVIEW

File: server/authentication/authentication.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  try {
+    const profile = await Parents.findOne({ email: userEmail });
+    cb(profile);
+  } catch (err) {
+    console.log(err);
+  }
+};
+
+
+exports.validtor = ({ email, password }, res) => {
```
Comment (misterrodger) [inline server/authentication/authentication.js]: maybe call it 'validator' ?

File: server/authentication/authentication.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          res.cookie('id', cb._id, { httpOnly: true });
+
+          res.json({ success: true, msg: 'Logged in successfully!' });
+        }
+      });
+    }
+  });
+};
+
+// validtor({ email: 'mail@gmail.com' });
```
Comment (tawfiknasser) [inline server/authentication/authentication.js]: Delete comments

File: server/authentication/authentication.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+const bcrypt = require('bcrypt');
+const mongoose = require('mongoose');
+const cookieparser = require('cookie');
+const Parents = require('../database/schemas/Parent.js');
+
+require('env2')('../../config.env');
+
+// const query = { email, password };
+// const router = express.Router();
+// router.get('/profile/:parentID', parentController.parent_findById);
```
Comment (tawfiknasser) [inline server/authentication/authentication.js]: Delete comments & unused code

File: client/src/components/login/login.js [hunk 4]
```diff
@@ -1,40 +1,57 @@
...
  };
 
  handleSubmit = event => {
-  event.preventDefault();
-  const data = JSON.stringify(this.state);
-  this.setState({ email: '', password: ''});
-  this.props.history.push('/events')
+   event.preventDefault();
+
+   const valid = async () => {
```
Comment (tawfiknasser) [inline client/src/components/login/login.js]: it's better to define it out side and not nest functions

File: client/src/components/login/login.js [hunk 5]
```diff
@@ -1,40 +1,57 @@
...
+         this.setState({text: data.msg})
+         if( data.success ) {
+           this.props.history.push('/events')
+         }
+       })
+     } catch (error) {
+       console.error(error)
+     }
+   }
+   valid()
```
Comment (tawfiknasser) [inline client/src/components/login/login.js]: add await for it

File: client/src/components/login/login.js [hunk 6]
```diff
@@ -1,40 +1,57 @@
...
 import './login.css'
+import axios from 'axios';
 
 export default class Login extends React.Component {
   state = {
-   email: '', password: ''
+   email: '', password: '', text: 'Please login to access our app'
  };
 
  handleSubmit = event => {
```
Comment (tawfiknasser) [inline client/src/components/login/login.js]: add async

File: server/authentication/authentication.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+const express = require('express');
+const bcrypt = require('bcrypt');
+const mongoose = require('mongoose');
+const cookieparser = require('cookie');
+const Parents = require('../database/schemas/Parent.js');
+
+require('env2')('../../config.env');
```
Comment (tawfiknasser) [inline server/authentication/authentication.js]: Is this the right path ?
Are you even using it

File: server/controllers/parentControllers.js [hunk 8]
```diff
@@ -52,3 +53,16 @@ exports.parent_findById = async (req, res) => {
...
   }
 };
+exports.parent_validator = (req, res) => {
+  try {
+    const checkuser = {
+      email: req.body.email,
+      password: req.body.password
+
+    };
+    authentication.validtor(checkuser, res);
```
Comment (tawfiknasser) [inline server/controllers/parentControllers.js]:
```suggestion
    authentication.validator(checkuser, res);
```

File: server/authentication/authentication.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+};
+
+
+exports.validator = ({ email, password }, res) => {
+  console.log(email, 'fae');
+  findUser(email, (cb) => {
+    if (cb === null || cb === undefined) {
+      res.json({ success: false, msg: "User doesn't exist!" });
+    } else {
+      console.log(cb.password, cb._id);
```
Comment (tawfiknasser) [inline server/authentication/authentication.js]: call back or object ?

**Pull request merged**